packages/amplify-codegen/src/commands/models.js

```diff
@@ -34,45 +36,55 @@ const modelgenFrontendToTargetMap = {
   introspection: 'introspection',
 };
 
+/**
+ * Return feature flag override values from the cli in the format --feature-flag:<feature flag name> <feature flag value>
+ * @param {!AmplifyContext} context the amplify runtime context
+ * @param {!string} flagName the feature flag name
+ * @returns {any | null} the raw value if found, else null
+ */
+const cliFeatureFlagOverride = (context, flagName) => context.parameters?.options?.[`feature-flag:${flagName}`];
+
 /**
  * Returns feature flag value, default to `false`
- * @param {!string} key feature flag id
+ * @param {!AmplifyContext} context the amplify runtime context
+ * @param {!string} flagName feature flag id
  * @returns {!boolean} the feature flag value
  */
-const readFeatureFlag = key => {
-  let flagValue = false;
+const readFeatureFlag = (context, flagName) => {
```

Comment:
With this defaulting to false, will every feature flag need to be passed? 

i.e. Outside of amplify project. `amplify codegen models --schema-path schema.graphql --target typescript`. This will have all feature flags as false?

I think if we are in a no init state we should check if the feature flag was passed in the CLI. If it is not passed we should return `undefined` and have the graphql-generator package handle the defaults for each feature flag.

Reply:
It only defaults to false if the call to FeatureFlags throws, so we'll still receive the 'New Project Default' value for each flag.